pkg/tcl/apitcl/v1/testworkflows.go

```diff
@@ -0,0 +1,205 @@
+// Copyright 2024 Testkube.
+//
+// Licensed as a Testkube Pro file under the Testkube Community
+// License (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//	https://github.com/kubeshop/testkube/blob/main/licenses/TCL.txt
+
+package v1
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/gofiber/fiber/v2"
+	"github.com/pkg/errors"
+
+	testworkflowsv1 "github.com/kubeshop/testkube-operator/api/testworkflows/v1"
+	"github.com/kubeshop/testkube/internal/common"
+	"github.com/kubeshop/testkube/pkg/api/v1/testkube"
+	mappers2 "github.com/kubeshop/testkube/pkg/tcl/workflowstcl/mappers"
+)
+
+func (s *apiTCL) ListTestWorkflowsHandler() fiber.Handler {
+	errPrefix := "failed to list test workflows"
+	return func(c *fiber.Ctx) (err error) {
+		workflows, err := s.getFilteredTestWorkflowList(c)
+		if err != nil {
+			return s.BadGateway(c, errPrefix, "client problem", err)
+		}
+		err = SendResourceList(c, "TestWorkflow", testworkflowsv1.GroupVersion, mappers2.MapTestWorkflowKubeToAPI, workflows.Items...)
+		if err != nil {
+			return s.InternalError(c, errPrefix, "serialization problem", err)
+		}
+		return
+	}
+}
+
+func (s *apiTCL) GetTestWorkflowHandler() fiber.Handler {
+	return func(c *fiber.Ctx) (err error) {
+		name := c.Params("id")
+		errPrefix := fmt.Sprintf("failed to get test workflow '%s'", name)
+		workflow, err := s.TestWorkflowsClient.Get(name)
+		if err != nil {
+			return s.ClientError(c, errPrefix, err)
+		}
+		err = SendResource(c, "TestWorkflow", testworkflowsv1.GroupVersion, mappers2.MapKubeToAPI, workflow)
+		if err != nil {
+			return s.InternalError(c, errPrefix, "serialization problem", err)
+		}
+		return
+	}
+}
+
+func (s *apiTCL) DeleteTestWorkflowHandler() fiber.Handler {
+	return func(c *fiber.Ctx) error {
+		name := c.Params("id")
+		errPrefix := fmt.Sprintf("failed to delete test workflow '%s'", name)
+		err := s.TestWorkflowsClient.Delete(name)
+		s.Metrics.IncDeleteTestWorkflow(err)
+		if err != nil {
+			return s.ClientError(c, errPrefix, err)
+		}
+		//skipExecutions := c.Query("skipDeleteExecutions", "")
+		//if skipExecutions != "true" {
+		//	// TODO: Delete Executions
+		//}
+		return c.SendStatus(http.StatusNoContent)
+	}
+}
+
+func (s *apiTCL) DeleteTestWorkflowsHandler() fiber.Handler {
+	errPrefix := "failed to delete test workflows"
+	return func(c *fiber.Ctx) error {
+		selector := c.Query("selector")
+		_, err := s.TestWorkflowsClient.List(selector)
+		if err != nil {
+			return s.BadGateway(c, errPrefix, "client problem", err)
+		}
+
+		err = s.TestWorkflowsClient.DeleteByLabels(selector)
+		if err != nil {
+			return s.ClientError(c, errPrefix, err)
+		}
+
+		//skipExecutions := c.Query("skipDeleteExecutions", "")
```

Comment:
should it be removed?

Reply:
I kept it to ensure that I'll not miss it when TestWorkflow executions will be there